src/OrasProject.Oras/Registry/Remote/Repository.cs

```diff
@@ -515,7 +522,7 @@ internal async Task ReferrersByApi(Descriptor descriptor, string? artifactType,
         // If artifactType is specified, apply any filters based on the artifact type
         if (!string.IsNullOrEmpty(artifactType))
         {
-            if (!response.Headers.TryGetValues(_headerOciFiltersApplied, out var values) || !Referrers.IsReferrersFilterApplied(values.FirstOrDefault(), artifactType))
+            if (!response.Headers.TryGetValues(_headerOciFiltersApplied, out var values) || !Referrers.IsReferrersFilterApplied(values.FirstOrDefault(), _filterTypeArtifactType))
```

Comment:
nit: Would it be more readable if we put the two conditions in separate lines?
```suggestion
            if (!response.Headers.TryGetValues(_headerOciFiltersApplied, out var values) 
            || !Referrers.IsReferrersFilterApplied(values.FirstOrDefault(), _filterTypeArtifactType))
```